### ISSUE TITLE
Add a "share" feature for other users

### ISSUE THREAD

**Issue by mastobuuthnaath** [nextcloud/tasks#72]
### Steps to reproduce
1. Create one User 
2. Create a second user
3. Create a task for the first User
4. Assign this Task to the second user

### Expected behaviour
This Task shall be visible by then by the second user

### Actual behaviour
This functionnality doesn't exist yet ?

**Comment by walhallaRV** [comment]
In the description of this app there is written that one can share tasks with other users. But I cant find this option to share? Where can I find it?

thanks a lot for your response!

Walhalla

**Comment by pvl-e** [comment]
I agree,please add the ability to do the task access to other users.

**Comment by elronzo** [comment]
+1
The ability to share tasks would be a great feature!

**Comment by tschiu** [comment]
Agree.  Sharing tasks would be great.  Am using shared calendars right now and it can be a bit cumbersome

**Comment by andreasjacobsen93** [comment]
This should be implemented so one could share a complete task list or only single tasks.

**Comment by cwmke** [comment]
This seems to be a more detailed duplicate of #14.

**Comment by Fitz22** [comment]
+1 
It's really important en enterprise to share a task with an other user.

**Comment by dartcafe** [comment]
Just another view regarding the sharing of tasks: I would like to assign users to tasks, so it is possible to delegate tasks to other team members. Is this the plan or is this another issue, like I think the author of #99 wanted to see. Or do I hassle things?

**Comment by zefir6** [comment]
Is there any plan to implement this functionality? It's kind of essential for shared tasks environment.

**Comment by SehLax** [comment]
the current way that sharing task lists works is explained in [#14 Share task lists](https://github.com/nextcloud/tasks/issues/14l).

here is a way how you can use this to assign tasks to people in your team:
1. create a new calendar in the calendar app and name it for example "to be done by team-memberXY"
2. still in calendar app, share the calendar with that user (with the option "can edit") (depending on the structure of your team, decide who can see&edit whose calendards/task lists)
3. now in task app, if you have a task and want to assign it to someone, simply move it (drag&drop) to this new calendar/task list

**Comment by Pokerkoffer** [comment]
What is the current status? This feature would be really nice :)

**Comment by GoetheG** [comment]
I am very interested in this function cause then I could really collaborate within nextcloud.

@SehLax: The key about widely spread software is it's user experience. This seems rather complicated to me. An unexperienced user would never find this solution easy to understand. Is there maybe a way to make all this a little bit easier? Just take a look at Open-XChange and their task functions. This is pretty simple to use.

Cheers

**Comment by dosch** [comment]
+1, we work in a small team and most kanban boards are over the top. The task app in NC is perfect. Just need to be able to assign it to team members....

**Comment by TruputiGalvotas** [comment]
Could I suggest to implement this simply as a category, but as a separate field on the task, so that on the backend it would be a category, but when you are on the "Assigned to User" field, you have a drop-down of users to select from to whom you can assign the task. Then when you have a shared task list between some members, then you can simply filter out tasks by user category. Or if using a category in the backend is a wrong way to do this then maybe it could be a separate column in the database - though I haven't looked at the Tasks app code and I don't know how it's actually implemented.

We are currently a small team of 3 people and we are using Categories as a way to assign the Task to a user, but having a dropdown of users list would be better.

![User_Task_Mockup](https://user-images.githubusercontent.com/19952587/54162914-f4bb2c80-445f-11e9-9f87-132ba199441f.png)

I would be willing to implement this myself with some guidance/mentoring from the nextcloud team if there would be a consensus on how this feature should be implemented.

**Comment by wdfee** [comment]
@TruputiGalvotas a user dropdown would be awesome, but the category solution doesn't fit for bigger teams. I think instead of a dropdown the Nextcloud typical text input with auto suggestions would be enough.
We are a whole school using Nextcloud for parents, teachers, students. We have classes as groups, e.g. one group for all parents of one class. We have rotating jobs like cleaning service, every weekend a different family (this is a private school with 36 students per class). So we make a Task list for cleaning service, each task is for one family. It would be great if we could assign these tasks to these families, that they get a reminder for "their" weekend.

**Comment by pwFoo** [comment]
It's possible to share tasks by share task (calendar) with users / groups. 

But task should be check permissions! Users mark tasks as done without permissions and it looks good for the user, but it's not syncing to task creator and other users. The user without change permission to the task calender should get an error message instead?

Assign tasks would be nice, following here.

**Comment by raimund-schluessler** [comment]
> But task should be check permissions! Users mark tasks as done without permissions and it looks good for the user, but it's not syncing to task creator and other users. The user without change permission to the task calender should get an error message instead?

Indeed, that is a bug. A user without editing permission should not be able to change a task's property locally. I created an issue for this: #522

**Comment by pcav** [comment]
share!=assign

**Comment by GoetheG** [comment]
Is this issue still open?

**Comment by pcav** [comment]
To me yes

**Comment by vincowl** [comment]
Same here !

**Comment by macmiranda** [comment]
+1

**Comment by steeve971** [comment]
+1 Any update ? It will be a great feature for tasks app... and it's very important for teammates to collaborate easily

**Comment by dukeh3** [comment]
After reading the CalDAV and ICAL specs I believe that I understand why this might be a bit trickier than expected. 

A VTODO lives in a calendar, this calendar has an owner. You cant simply share a task, because its not how its is done. A task i my calendar is not connected to a task in your calendar. Ofcoure you could make a tast with ID T1 live in several calendars and if the task gets updated in one also update the task in all other places. I am not sure how the different software out there would react to this. But I assume that CalDAV does handle this.

I would assume that the way this would be done is that you create a project calendar and then create the task there, and the use the ATTENDEE and ORGANIZER fields to indicate who is the owner and who is interested in the task. 

https://icalendar.org/iCalendar-RFC-5545/3-6-2-to-do-component.html

https://icalendar.org/iCalendar-RFC-5545/3-8-4-1-attendee.html

I assume that a good point to start would be to see how this is implemented in the EVENT component in the calendar. For there you can add people to an event.

**Comment by szaimen** [comment]
You might also have a look at the Nextcloud Deck app. I heard that it is a caldav client nowadays, too.

**Comment by annievoss** [comment]
Is there any update on this? It would be nice to be able to assign a singe task to another user, like cards in Deck App.

**Comment by jogood** [comment]
Hi there, I too would like to be able to assign tasks to user.

I'm currently trying to implement it. I'd really like to have some help from someone who understand how the task plugin work. The Task plugin is great but it lacks what seems to be very simple features that would make it even better.

**Comment by x-tk** [comment]
Would be really great if this was added, surprised it's not an option yet still

**Comment by meonkeys** [comment]
> Would be really great if this was added, surprised it's not an option yet still

I'm not surprised.

Just add a thumbs up to the original description. Saves others having to scroll past "me too" comments with no additional information.

**Comment by Zaferfe** [comment]
their duties can be assigned to another personnel,
There is no file attachment area.
For example, the manager should be able to assign tasks to the personnel, prepare a sales report for me, assign the task to the personnel, and put them in the file area by doing that task.

**Comment by bpbiszkopt** [comment]
Any update on this future? Is there any possibility to add option even as @TruputiGalvotas suggested?

**Comment by punkyard** [comment]
> tasks can be assigned to users, Users can see assigned tasks in task list

Hi I can't find this assignment field, could you guide me to there please?

**Comment by punkyard** [comment]
WORK AROUND
waiting for task assignment, tags could be used as a work around:
type in the tag the name of the user, press Enter, their name will appear as a tag on the task line

![Uploading Capture d’écran 2024-01-17 à 09.14.55.png…]()

**Comment by lhyphendixon** [comment]
How on earth was this not one of the first features for Tasks?  If there are two things I want to do; 1. Write down task 2. Assign it.  I'm flabbergasted. 7 years?

**Comment by raimund-schluessler** [comment]
> How on earth was this not one of the first features for Tasks?  If there are two things I want to do; 1. Write down task 2. Assign it.  I'm flabbergasted. 7 years?

The issue is free to be worked on by everyone. Pull requests are always welcome, thank you!

**Comment by adam42a** [comment]
The optimal user functionality from my POV would be to allow people to share a task list with a group. Then anyone from that group can be added to the task as collaborators or responsible for the task completion. 

People not in the group can't see the task list. 

I didn't really understand much of the calendar-related functionality requests and to be honest, with all the task list platforms I've used in all the projects I've been involved with, I mostly didn't use calendar functionality, preferring to use the task position in the list as a sign of its priority or urgency. 

I hope you devs who might work on this like the suggestion. I am coming to Tasks as it's the best candidate for my team to replace both Trello boards and Notion lists which we can't own properly and have to pay extra for. Tasks really could fill that niche.

**Comment by zaferfe** [comment]
Good day, I also think like this, I provide IT support service. There are
100 users, if the user's printer, keyboard, mouse or a program does not
work, they can send me a task notification, that user can only see what
they have written, I can see all the problems. When I look at the list, it
is not visible which user has defined the task, it is not known who has
which problem.

Adam ***@***.***>, 9 Şub 2025 Paz, 22:33 tarihinde şunu yazdı:

**Comment by h4mr3r** [comment]
Any update? Will this feature be implemented into this app?

**Comment by kneckergit** [comment]
I also second this request. It is most vital for teamwork.

**Comment by ecc86** [comment]
This is actually a super important feature for meetings one would perfectly well have in Nextcloud Talk: noting some tasks and assigning them right away to a team member. So +1
